sorts/quick_sort.ts

```diff
@@ -56,17 +66,15 @@ export const QuickSort = (
   left: number = 0,
   right: number = array.length - 1
 ) => {
-  let index;
-
   if (array.length > 1) {
-    index = partition(array, left, right);
+    const index = partition(array, left, right);
 
     if (left < index - 1) {
       QuickSort(array, left, index - 1);
     }
 
-    if (index < right) {
-      QuickSort(array, index, right);
+    if (index + 1 < right) {
```

Comment:
This change is needed, because, if we randomize the pivot, then, hypothetically (but statistically unlikely), we might get that the pivot is always the left-most element. In this case, `QuickSort` will fail, as we would call the same `QuickSort` over and over again until we entirely pollute the call stack.